apps/back/src/Command/ViewBalanceCommand.php

```diff
@@ -18,47 +17,45 @@
 )]
 class ViewBalanceCommand extends Command
 {
+    public function __construct(private ManagerRegistry $doctrine)
+    {
+        parent::__construct();
+    }
+
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $andre = new User('Andre','Toto','andre@gmail.com');
-        $caroline = new User('Caroline', 'Titi','caro@gmail.com');
-        $benoit = new User('Benoit', 'Tutu', 'benoit@gmail.com');
-        $users = [$andre,$caroline,$benoit];
-        
-        $expense1 = new Expense('essence', new DateTime('20 march 2023'), 'plein d\'essence pour aller au match', 30, [$andre, $caroline, $benoit], $andre);
-        $expense2 = new Expense('souvenir', new DateTime('21 march 2023'), 'souvenir de notre tournoi à Caen', 20, [$andre, $benoit], $andre);
-        $expense3 = new Expense('bar', new DateTime('24 march 2023'), 'tournee', 15, [$caroline, $benoit], $caroline);
-        $expense4 = new Expense('Materiel', new DateTime('25 march 2023'), 'nouveau materiel', 60, [$andre, $caroline, $benoit], $benoit);
-        $expenses = [$expense1,$expense2,$expense3,$expense4];
+        $groupRepository = $this->doctrine->getRepository(Group::class);
+        $group = $groupRepository->findOneBy(['name' => 'bad']);
+        $members = $group->getMembers();
+        $groupExpenses = $group->getExpensesOfTheGroup();
+        $groupBalance = [];
+
+        foreach ($members as $member) {
+            $groupBalance[$member->getId()]= 0;
+        }
```

Comment:
I think you don't need to iterate two times on `members`. 

You should add it to your foreach line 42 when the member don't have any expense yet.

Reply:
I think if I add it to the foreach it will be re-initialize at 0 everytime I enter in the loop `foreach ($groupExpenses as $groupExpense),` ?